System/Vector.h

```diff
@@ -17,45 +17,37 @@ namespace RTE {
 
 		SerializableOverrideMethods
 
-		float m_X; //!< X value of this vector.
-		float m_Y; //!< Y value of this vector.
+		float m_X = 0.0F; //!< X value of this vector.
+		float m_Y = 0.0F; //!< Y value of this vector.
 
 #pragma region Creation
 		/// <summary>
-		/// Constructor method used to instantiate a Vector object.
+		/// Constructor method used to instantiate a Vector object with values (0, 0).
 		/// </summary>
-		Vector() { Clear(); }
+		Vector() {};
 
 		/// <summary>
 		/// Constructor method used to instantiate a Vector object from X and Y values.
 		/// </summary>
 		/// <param name="inputX">Float defining the initial X value of this Vector.</param>
 		/// <param name="inputY">Float defining the initial Y value of this Vector.</param>
-		Vector(float inputX, float inputY) { Create(inputX, inputY); }
+		Vector(const float inputX, const float inputY) :
+			m_X(inputX),
+			m_Y(inputY) {};
 
 		/// <summary>
 		/// Copy constructor method used to instantiate a Vector object identical to an already existing one.
 		/// </summary>
 		/// <param name="reference">A Vector object which is passed in by reference.</param>
-		Vector(const Vector &reference) { Create(reference.m_X, reference.m_Y); }
-
-		/// <summary>
-		/// Makes the Vector object ready for use.
-		/// </summary>
-		/// <param name="inputX">Float defining the initial X value of this Vector.</param>
-		/// <param name="inputY">Float defining the initial Y value of this Vector.</param>
-		/// <returns>An error return value signaling success or any particular failure. Anything below 0 is an error signal.</returns>
-		int Create(float inputX, float inputY) { m_X = inputX; m_Y = inputY; return 0; }
+		Vector(const Vector &reference) = default;
 #pragma endregion
 
-#pragma region Destruction
+#pragma region Getters and Setters
 		/// <summary>
 		/// Sets both the X and Y of this Vector to zero.
 		/// </summary>
-		void Reset() { m_X = m_Y = 0.0F; }
-#pragma endregion
+		void Reset() { m_X = 0.0F; m_Y = 0.0F; }
```

Comment:
I'm fine with it being `Reset`, just put it back into `Destruction` and slap `override` on it because it's supposed to be inherited from `Serializable`.

And FYI, renaming methods on the cpp side doesn't have to break them for lua.
`.def("Reset", &Vector::Reset)`
This is the binding, you can rename it to whatever you want on either side.
If you were to rename the cpp method but wanted to keep existing lua compatibility, you'd just do
`.def("Reset", &Vector::NewMethodName)`
If you wanted to destroy existing lua without renaming the cpp method, you'd just do
`.def("HaHaIBrokeLua", &Vector::Reset)`

Reply:
Override 👍 
And I'll put it in destruction, but I suggest to rethink what goes under it.
Reset() doesn't destroy anything so to me it doesn't sound fitting to put it under Destruction.

The lua stuff slipped my mind thanks for the reminder.

Reply:
Yeah I know it's not the most accurate category to put it under but we'll keep it there for now for consistency.